ReminderApp-OBJC/ReminderApp-OBJC/Classes/Controller/CDTopicTableViewController.m

```diff
@@ -0,0 +1,223 @@
+//  Copyright © 2016 Catalin David. All rights reserved.
+//
+
+#import "CDTopicTableViewController.h"
+#import "AppDelegate.h"
+#import "CDTopic.h"
+#import "CDReminderTableViewController.h"
+
+#pragma mark - Class extension
+
+@interface CDTopicTableViewController () <NSFetchedResultsControllerDelegate>
+
+@property (nonatomic, weak) IBOutlet UIBarButtonItem *editButton;
+@property (nonatomic, strong) NSFetchedResultsController *fetchedResultsController;
+@property (nonatomic, strong) NSManagedObjectContext *managedContext;
+
+@end
+
+#pragma mark - Class implementation
+
+@implementation CDTopicTableViewController
+
+- (void)viewDidLoad {
+	
+	[super viewDidLoad];
+	
+	self.editButton.title = @"Edit";
+	self.title = @"Topics";
+	
+	self.managedContext = ((AppDelegate*)[UIApplication sharedApplication].delegate).managedObjectContext;
+	
+	NSError *error;
+	if (![self.fetchedResultsController performFetch:&error]) {
+		NSLog(@"An error occured: %@", error);
+	}
+	
+}
+
+-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+	
+	if ([segue.identifier isEqualToString:@"segueTopicControllerToReminderController"]) {
+		CDReminderTableViewController *destinationController = ((CDReminderTableViewController *)segue.destinationViewController);
+		destinationController.topic = ((CDTopic *) [self.fetchedResultsController objectAtIndexPath:self.tableView.indexPathForSelectedRow]);
+	}
+	
+}
+
+#pragma mark - Specific action buttons
+
+- (IBAction)editTapped:(id)sender {
+
+	if (self.tableView.editing) {
+		self.editButton.title = @"Edit";
+		self.tableView.editing = NO;
+	} else {
+		self.editButton.title = @"Back";
+		self.tableView.editing = YES;
+	}
+	
+}
+
+- (IBAction)addTapped:(id)sender {
+	
+	UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"New Topic"
+																   message:@"Add a new topic'"
+															preferredStyle:UIAlertControllerStyleAlert];
+	
+	UIAlertAction *saveAction = [UIAlertAction actionWithTitle:@"Save"
+														 style:UIAlertActionStyleDefault
+													   handler:^(UIAlertAction	*action) {
+														   
+														   NSString *textField = alert.textFields.firstObject.text;
+														   [self saveTask:textField];
+													   } ];
+	
+	UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"Cancel"
+														   style:UIAlertActionStyleDefault
+														 handler:nil];
+	
+	[alert addTextFieldWithConfigurationHandler:nil];
+	[alert addAction:saveAction];
+	[alert addAction:cancelAction];
+	
+	[self presentViewController:alert animated:YES completion:nil];
+	
+}
+
+#pragma mark - Helper func for addTapped
+
+- (void) saveTask:(NSString *)topicName {
+	
+	CDTopic *topic = [NSEntityDescription insertNewObjectForEntityForName:@"Topic" inManagedObjectContext:self.managedContext];
+	topic.title = topicName;
+	
+	NSError *error;
+	if (![self.managedContext save:&error]) {
+		NSLog(@"An error occured: %@", error);
```

Comment:
pune mesaje de eroare mai specifice, ca atunci cand o sa iti creasca aplicatia o sa ai o gramada de msaje "An error occured:" si nu sti de ce cod apartiene. Ex "Could not save the Topic objext : Error : %@".
Valabil la toate erorile de acest gen.